domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/induction/service/InductionScheduleService.kt

```diff
@@ -61,81 +88,83 @@ class InductionScheduleService(
     prisonNumber: String,
     newStatus: InductionScheduleStatus,
     exemptionReason: String?,
-  ) {
+  ): InductionSchedule {
     val inductionSchedule = inductionSchedulePersistenceAdapter.getInductionSchedule(prisonNumber)
       ?: throw InductionScheduleNotFoundException(prisonNumber)
 
     // Validate the status transition
     inductionScheduleStatusTransitionValidator.validate(prisonNumber, inductionSchedule.scheduleStatus, newStatus)
 
-    when {
+    return when {
       newStatus == InductionScheduleStatus.EXEMPT_SYSTEM_TECHNICAL_ISSUE -> {
-        updateInductionScheduleFollowingSystemTechnicalIssue(inductionSchedule, exemptionReason, prisonNumber)
+        updateInductionScheduleFollowingSystemTechnicalIssue(inductionSchedule, exemptionReason)
       }
 
       newStatus.isExemptionOrExclusion() -> {
-        updateExemptStatus(inductionSchedule, newStatus, exemptionReason, prisonNumber)
+        exemptInductionSchedule(inductionSchedule, newStatus, exemptionReason)
       }
 
       else -> {
-        updateScheduledStatus(inductionSchedule, prisonNumber)
+        val adjustedInductionDueDate = inductionScheduleDateCalculationService.calculateAdjustedInductionDueDate(inductionSchedule)
+        rescheduleInductionSchedule(inductionSchedule, adjustedInductionDueDate)
       }
     }
   }
 
-  private fun updateExemptStatus(
+  private fun exemptInductionSchedule(
```

Comment:
Some minor refactoring of these private methods, of their names, but also to return the InductionSchedule they have just updated; which means we can return it all the way up the stack (which I wanted to in my new re-schedule method)